src/games/strategy/triplea/ai/proAI/data/ProMoveOptions.java

```diff
@@ -0,0 +1,116 @@
+package games.strategy.triplea.ai.proAI.data;
+
+import games.strategy.engine.data.PlayerID;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.ai.proAI.util.ProBattleUtils;
+import games.strategy.triplea.ai.proAI.util.ProUtils;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.util.Match;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class ProMoveOptions {
+
+  private final ProUtils utils;
+  private final ProBattleUtils battleUtils;
+  private final Map<Territory, ProTerritory> maxMoveMap;
+  private final Map<Territory, List<ProTerritory>> moveMaps;
+
+  public ProMoveOptions(final ProUtils utils, final ProBattleUtils battleUtils) {
+    this.utils = utils;
+    this.battleUtils = battleUtils;
+    maxMoveMap = new HashMap<Territory, ProTerritory>();
+    moveMaps = new HashMap<Territory, List<ProTerritory>>();
+  }
+
+  public ProMoveOptions(final ProUtils utils, final ProBattleUtils battleUtils,
+      final List<Map<Territory, ProTerritory>> moveMapList, final PlayerID player, final boolean isAttacker) {
+    this(utils, battleUtils);
+    populateMaxMoveMap(moveMapList, player, isAttacker);
+    populateMoveMaps(moveMapList);
```

Comment:
Hmm, maybe you do want to be able to instantiate an empty version of this object, I haven't checked, but having the populate methods return something rather than 'void' would be an improvement.